mediapipe/examples/android/src/java/com/google/mediapipe/apps/instantmotiontracking/MainActivity.java

```diff
@@ -167,36 +380,66 @@ private void setupPreviewDisplayView() {
-                        // based on the size of the SurfaceView that contains the display.
-                        Size viewSize = new Size(width, height);
-                        Size displaySize = cameraHelper.computeDisplaySizeFromViewSize(viewSize);
-                        boolean isCameraRotated = cameraHelper.isCameraRotated();
-
-                        // Connect the converter to the camera-preview frames as its input (via
-                        // previewFrameTexture), and configure the output width and height as the computed
-                        // display size.
-                        converter.setSurfaceTextureAndAttachToGLContext(
-                                previewFrameTexture,
-                                isCameraRotated ? displaySize.getHeight() : displaySize.getWidth(),
-                                isCameraRotated ? displaySize.getWidth() : displaySize.getHeight());
-                      }
-
-                      @Override
-                      public void surfaceDestroyed(SurfaceHolder holder) {
-                        processor.getVideoSurfaceOutput().setSurface(null);
-                      }
-                    });
+        .getHolder()
+        .addCallback(
+            new SurfaceHolder.Callback() {
+              @Override
+              public void surfaceCreated(SurfaceHolder holder) {
+                processor.getVideoSurfaceOutput().setSurface(holder.getSurface());
+              }
+
+              @Override
+              public void surfaceChanged(SurfaceHolder holder, int format, int width, int height) {
+                // (Re-)Compute the ideal size of the camera-preview display (the area that the
+                // camera-preview frames get rendered onto, potentially with scaling and rotation)
+                // based on the size of the SurfaceView that contains the display.
+                Size viewSize = new Size(width, height);
+                Size displaySize = cameraHelper.computeDisplaySizeFromViewSize(viewSize);
+                boolean isCameraRotated = cameraHelper.isCameraRotated();
+
+                // Connect the converter to the camera-preview frames as its input (via
+                // previewFrameTexture), and configure the output width and height as the computed
+                // display size.
+                converter.setSurfaceTextureAndAttachToGLContext(
+                    previewFrameTexture,
+                    isCameraRotated ? displaySize.getHeight() : displaySize.getWidth(),
+                    isCameraRotated ? displaySize.getWidth() : displaySize.getHeight());
+              }
+
+              @Override
+              public void surfaceDestroyed(SurfaceHolder holder) {
+                processor.getVideoSurfaceOutput().setSurface(null);
+              }
+            });
+  }
+
+  private void prepareDemoAssets() {
+    AndroidAssetUtil.initializeNativeAssetManager(this);
```

Comment:
Looks like this was done above too-- can we remove it from one of the locations?